src/global-firestore-data.ts

```diff
@@ -40,8 +42,10 @@ const editSemesters = (
   updater: (oldSemesters: readonly FirestoreSemester[]) => readonly FirestoreSemester[]
 ): void => {
   const newSemesters = updater(store.state.semesters);
-  store.commit('setSemesters', newSemesters);
-  semestersCollection.doc(store.state.currentFirebaseUser.email).set({ semesters: newSemesters });
+  semestersCollection
+    .doc(store.state.currentFirebaseUser.email)
+    .set({ semesters: newSemesters })
+    .then(() => store.commit('setSemesters', newSemesters));
```

Comment:
Is this change necessary? I think it might be causing the two bugs:

Dragging a course to a different semester deletes the course. First tab is this PR, second tab is dev staging:

https://user-images.githubusercontent.com/54298311/117380056-ebdf0d80-aea6-11eb-854b-a124a3304bae.mov

When I add a course to a requirement with a checkerWarning (like Technical Electives) the course doesn't show. First tab is this PR, second is dev:

https://user-images.githubusercontent.com/54298311/117380370-8b9c9b80-aea7-11eb-98e8-458cf5fc86f1.mov

Reply:
Oooh yeah thanks for finding the bug and also finding the relevant code!! Nah the code didn't need to change, it's an artifact of my brain fart regarding the listeners.